src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        linkLastElement(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size) {
+            linkLastElement(value);
+        } else {
+            linkBeforeElement(value, searchForNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.isEmpty()) {
+            return;
+        }
+        Node<T> previous = last;
+        T[] elements = (T[]) list.toArray();
+        for (T element : elements) {
+            Node<T> newNode = new Node<>(previous, element, null);
+            if (previous == null) {
+                first = newNode;
+            } else {
+                previous.next = newNode;
+                previous = newNode;
+            }
+            last = previous;
+        }
+        size += elements.length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return searchForNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> temp = searchForNode(index);
+        T oldValue = temp.item;
+        temp.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(searchForNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> temp = first; temp != null; temp = temp.next) {
+            if (object != null && object.equals(temp.item) || temp.item == object) {
```

Comment:
The condition `temp.item == object` should be checked first to avoid a potential `NullPointerException` when `object` is null.